TCAT/AppDelegate.swift

```diff
@@ -160,6 +160,79 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         alertController.addAction(UIAlertAction(title: "OK", style: .cancel, handler: nil))
         UIApplication.shared.keyWindow?.presentInApp(alertController)
     }
+
+    /// When app is opened via URL
+    // URLs for testing
+    // BusStop: ithaca-transit://getRoutes?lat=42.442558&long=-76.485336&stopName=Collegetown
+    // PlaceResult: ithaca-transit://getRoutes?lat=42.44707979999999&long=-76.4885196&destinationName=Hans%20Bethe%20House
+    func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey: Any] = [:]) -> Bool {
+
+        let rootVC = HomeViewController()
+        let navigationController = CustomNavigationController(rootViewController: rootVC)
+        self.window = UIWindow(frame: UIScreen.main.bounds)
+        self.window!.rootViewController = navigationController
+        self.window?.makeKeyAndVisible()
+
+        let urlComponents = NSURLComponents(url: url, resolvingAgainstBaseURL: false)
+        let items = (urlComponents?.queryItems)! as [NSURLQueryItem]
+        var stop: BusStop
+
+        // if the URL is from the getRoutes intent
+        if url.absoluteString.contains("getRoutes") {
+            var latitude: CLLocationDegrees?
+            var longitude: CLLocationDegrees?
+            var stopName: String?
+            let optionsVC = RouteOptionsViewController()
+
+            // get parameters from the URL
+            for (index, element) in items.enumerated() {
+                switch index {
+                case 0:
+                    if let propertyValue = element.value, let lat = Double(propertyValue) {
+                        latitude = lat
+                    }
+                case 1:
+                    if let propertyValue = element.value, let long = Double(propertyValue) {
+                        longitude = long
+                    }
+                case 2:
+                    if let propertyValue = element.value {
+                        stopName = propertyValue
+                    }
+                default:
+                    return false
+                }
+            }
+
+            if let latitude = latitude, let longitude = longitude, let stopName = stopName {
+                stop = BusStop(name: stopName, lat: latitude, long: longitude)
+                optionsVC.searchTo = stop
+                navigationController.pushViewController(optionsVC, animated: false)
+                return true
+            }
+        }
+
+        return false
+    }
+
+    func application(_ application: UIApplication, continue userActivity: NSUserActivity, restorationHandler: @escaping ([UIUserActivityRestoring]?) -> Void) -> Bool {
+
+        if #available(iOS 12.0, *) {
+            if let intent = userActivity.interaction?.intent as? GetRoutesIntent,
```

Comment:
nitpick: Hit return after the if on L221, to make all the `let` statements have the same indentation. Reads a bit better